### ISSUE TITLE
[RLlib Trajectory View API] Trajectory View API can replace frame-stacking setup in Models (e.g. Atari).

### ISSUE THREAD

**Issue by sven1977** [ray-project/ray#10882]
The Trajectory View API is able to replace the need for the frame-stacking option in the models through the `shift` parameter in the `ViewRequirement` objects, e.g.:
```
my_model_view_reqs = {
    "obs": ViewRequirement(data_col="obs", shift=[-3, -2, -1, 0], ...)
}
```

**Comment by mgerstgrasser** [comment]
Is it possible to do other parts of the usual deepmind-processing inside the model as well, e.g. max-and-skip? We have some multiagent settings where we want to apply different processing for different agents. It's possible to do that through wrappers, but doing it inside the model would be a lot cleaner.

**Comment by simonsays1980** [comment]
@mgerstgrasser This should be possible to do inside the model as shown here (copied from the [documentation](https://docs.ray.io/en/latest/rllib/rllib-models.html#more-examples-for-building-custom-models)):

```python


class FrameStackingCartPoleModel(TFModelV2):
    """A simple FC model that takes the last n observations as input."""

    def __init__(
        self, obs_space, action_space, num_outputs, model_config, name, num_frames=3
    ):
        super(FrameStackingCartPoleModel, self).__init__(
            obs_space, action_space, None, model_config, name
        )

        self.num_frames = num_frames
        self.num_outputs = num_outputs

        # Construct actual (very simple) FC model.
        assert len(obs_space.shape) == 1
        obs = tf.keras.layers.Input(shape=(self.num_frames, obs_space.shape[0]))
        obs_reshaped = tf.keras.layers.Reshape([obs_space.shape[0] * self.num_frames])(
            obs
        )
        rewards = tf.keras.layers.Input(shape=(self.num_frames))
        rewards_reshaped = tf.keras.layers.Reshape([self.num_frames])(rewards)
        actions = tf.keras.layers.Input(shape=(self.num_frames, self.action_space.n))
        actions_reshaped = tf.keras.layers.Reshape([action_space.n * self.num_frames])(
            actions
        )
        input_ = tf.keras.layers.Concatenate(axis=-1)(
            [obs_reshaped, actions_reshaped, rewards_reshaped]
        )
        layer1 = tf.keras.layers.Dense(256, activation=tf.nn.relu)(input_)
        layer2 = tf.keras.layers.Dense(256, activation=tf.nn.relu)(layer1)
        out = tf.keras.layers.Dense(self.num_outputs)(layer2)
        values = tf.keras.layers.Dense(1)(layer1)
        self.base_model = tf.keras.models.Model([obs, actions, rewards], [out, values])
        self._last_value = None

        self.view_requirements["prev_n_obs"] = ViewRequirement(
            data_col="obs", shift="-{}:0".format(num_frames - 1), space=obs_space
        )
        self.view_requirements["prev_n_rewards"] = ViewRequirement(
            data_col="rewards", shift="-{}:-1".format(self.num_frames)
        )
        self.view_requirements["prev_n_actions"] = ViewRequirement(
            data_col="actions",
            shift="-{}:-1".format(self.num_frames),
            space=self.action_space,
        )

    def forward(self, input_dict, states, seq_lens):
        obs = tf.cast(input_dict["prev_n_obs"], tf.float32)
        rewards = tf.cast(input_dict["prev_n_rewards"], tf.float32)
        actions = one_hot(input_dict["prev_n_actions"], self.action_space)
        out, self._last_value = self.base_model([obs, actions, rewards])
        return out, []

    def value_function(self):
        return tf.squeeze(self._last_value, -1)
```

**Comment by Rohan138** [comment]
Closing this as solved, see example in #20827
